src/prime_differentiation.jl

```diff
@@ -0,0 +1,50 @@
+immutable PrimeDiff{N, F}
+    f::F
+end
+
+@generated function (::Type{PrimeDiff{N}}){N, F}(f::F)
+    ex = :(PrimeDiff{1, F}(f))
+    for i in 2:N
+        ex = :(_add_prime($ex))
+    end
+    return quote
+        $(Expr(:meta, :inline))
+        $ex
+    end
+end
+
+@generated function _add_prime{N}(p::PrimeDiff{N})
+    return quote
+        $(Expr(:meta, :inline))
+        PrimeDiff{$(N+1), typeof(p)}(p)
+    end
+end
+@inline Base.ctranspose{N}(p::PrimeDiff{N}) = _add_prime(p)
+
+@inline (p::PrimeDiff)(x) = gradient(p.f, x)
+
+function use_prime_diff()
+    @inline Base.ctranspose{F <: Function}(f::F) = Tensors.PrimeDiff{1}(f)
```

Comment:
This will just overwrite the method from Calculus, so after this the Calculus version is gone. This is also type piracy (function piracy?) and I think it is very strange that Calculus.jl are allowed to define this in the first place. So I'm not sure about this...